tests/tests_fabric/test_fabric.py

```diff
@@ -1203,7 +1219,7 @@ def test_verify_launch_called():
 @pytest.mark.parametrize(
     "kwargs",
     [
-        {},
+        pytest.param({}, marks=pytest.mark.xfail(raises=RuntimeError, match="Unsupported")),
```

Comment:
Simply doing `torch.is_grad_enabled() and not torch.is_inference_mode_enabled()` already graph breaks dynamo.

This PR adds graph break sources to the `_FabricModule`. However, there are already several issues so it's not like this proposal is the only issue.

Reply:
Note to self: https://github.com/pytorch/pytorch/issues/115970#issuecomment-1861277895